pkg/common/k8s_utils.go

```diff
@@ -147,3 +155,18 @@ func GetLabel(obj metav1.Object, key string) string {
 	}
 	return obj.GetLabels()[key]
 }
+
+func GenerateClusterID(ctx context.Context, c client.Client) (error, string) {
+	//Already calculated? return it
+	if clusterID != "" {
+		return nil, clusterID
+	}
+
+	ns := &corev1.Namespace{}
+	err := c.Get(ctx, client.ObjectKey{Name: clusterIDNamespace}, ns)
+	if err != nil {
+		return err, ""
+	}
+	clusterID = multicluster.ToBase36HashLen(string(ns.UID), clusterIDLength)
```

Comment:
Probably better to move `ToBase36HashLen` to a more common package if we are using it outside of `multicluster`.

Reply:
good idea